plugins/modules/kutils.py

```diff
@@ -0,0 +1,372 @@
+                principal=cmd_runner_fmt.as_opt_val('-p'),
+            )
+        )
+
+        self.klist = CmdRunner(
+            module,
+            command='klist',
+            arg_formats=dict(
+                show_list=cmd_runner_fmt.as_bool('-l'),
+            )
+        )
+
+    def exec_kinit(self):
+        params = dict(self.module.params)
+        with self.kinit(
+            "lifetime start_time renewable forwardable proxiable address_restricted anonymous "
+            "canonicalization enterprise renewal validate keytab keytab_path cache_name principal",
+            check_rc=True,
+            data=self.password,
+        ) as ctx:
+            rc, out, err = ctx.run(**params)
+        return out
+
+    def exec_kdestroy(self):
+        params = dict(self.module.params)
+        with self.kdestroy(
+            "kdestroy_all cache_name principal",
+            check_rc=True
+        ) as ctx:
+            rc, out, err = ctx.run(**params)
+        return out
+
+    def exec_klist(self, show_list):
+        # Use chech_rc = False because
+        # If no tickets present, klist command will always return rc = 1
+        params = dict(show_list=show_list)
+        with self.klist(
+            "show_list",
+            check_rc=False
+        ) as ctx:
+            rc, out, err = ctx.run(**params)
+        return rc, out, err
+
+    def check_ticket_present(self):
+        ticket_present = True
+        show_list = False
+
+        if not self.principal and not self.cache_name:
+            rc, out, err = self.exec_klist(show_list)
+            if rc != 0:
+                ticket_present = False
+        else:
+            show_list = True
+            rc, out, err = self.exec_klist(show_list)
+            if self.principal and self.principal not in str(out):
+                ticket_present = False
+            if self.cache_name and self.cache_name not in str(out):
+                ticket_present = False
+
+        return ticket_present
+
+
+def main():
+    arg_spec = dict(
+        principal=dict(type='str'),
+        password=dict(type='str', no_log=True),
+        state=dict(default='present', choices=['present', 'absent']),
+        kdestroy_all=dict(type='bool'),
+        cache_name=dict(type='str', fallback=(env_fallback, ['KRB5CCNAME'])),
+        lifetime=dict(type='str'),
+        start_time=dict(type='str'),
+        renewable=dict(type='str'),
+        forwardable=dict(type='bool'),
+        proxiable=dict(type='bool'),
+        address_restricted=dict(type='bool'),
+        anonymous=dict(type='bool'),
+        canonicalization=dict(type='bool'),
+        enterprise=dict(type='bool'),
+        renewal=dict(type='bool'),
+        validate=dict(type='bool'),
+        keytab=dict(type='bool'),
+        keytab_path=dict(type='path'),
+    )
+    module = AnsibleModule(
+        argument_spec=arg_spec,
+        supports_check_mode=True,
+        required_by={
+            'keytab_path': 'keytab'
+        },
```

Comment:
The docs for `keytab_path` state that when it is provided, then `keytab` must be `True`. This `required_by` will not guarantee you that - it will only guarantee that `keytab` is not `None`. Maybe the module needs to check and raise and exception if `keytab_path is not None and keytab is not True`.